tensorflow/python/training/learning_rate_decay.py

```diff
@@ -77,10 +78,173 @@ def exponential_decay(learning_rate, global_step, decay_steps, decay_rate,
+    epoch: A Python number.
+      Global step to use for the decay computation.  Must not be negative.
+    decay_rate: A Python number.  The decay rate.
+    name: String.  Optional name of the operation.  Defaults to
+      'ExponentialTimeDecay'
+
+  Returns:
+    A scalar `Tensor` of the same type as `learning_rate`.  The decayed
+    learning rate.
+  """
+
+  with ops.op_scope([initial_lr, epoch, decay_rate],
+                   name, "InverseTimeDecay") as name:
+    initial_lr = ops.convert_to_tensor(initial_lr, name="learning_rate")
+    decay_rate = cast(decay_rate, initial_lr.dtype)
+    const = cast(constant(1), initial_lr.dtype)
+    denom = add(const, mul(decay_rate, epoch))
+    return div(initial_lr, denom, name=name)
+
+def step_time_decay(learning_rate, epoch, num_steps_for_decay, decay_rate=0.5,
+               name=None):
+  """Applies step decay to the learning rate.
+
+  When training a model, it is often recommended to lower the learning rate as
+  the training progresses.  This function applies a step decay function
+  to a provided initial learning rate.  It requires a `global_step` value to
+  compute the decayed learning rate.  You can just pass a python number
+  that you increment at each training step.
+
+  The function returns the decayed learning rate.  It is computed as:
+
+  ```python
+  if global_step and global_step % num_steps_for_decay == 0:
+    decayed_learning_rate = learning_rate * decay_rate
+  ```
+  ```
+
+  Example: decay every 1000 steps with a base of 0.5:
+
+  ```python
+  ...
+  global_step = tf.Variable(0, trainable=False)
+  starter_learning_rate = 0.1
+  learning_rate = tf.train.exponential_decay(starter_learning_rate, global_step,
+                                             1000, 0.5)
+  # Passing global_step to minimize() will increment it at each step.
+  learning_step = (
+      tf.GradientDescentOptimizer(learning_rate)
+      .minimize(...my loss..., global_step=global_step)
+  )
+  ```
+
+  Args:
+    learning_rate: A scalar `float32` or `float64` `Tensor` or a
+      Python number.  The initial learning rate.
+    global_step: A Python number.
+      Global step to use for the decay computation.  Must not be negative.
+    num_steps_for_decay: A scalar `int32` or `int64` `Tensor` or a Python
+      number.
+      Must be positive.  See the decay computation above.
+    decay_rate: A scalar `float32` or `float64` `Tensor` or a
+      Python number.  The decay rate.
+    name: String.  Optional name of the operation.  Defaults to 'StepDecay'
+
+  Returns:
+    A scalar `Tensor` of the same type as `learning_rate`.  The decayed
+    learning rate.
+  """
+  with ops.op_scope([learning_rate, epoch, num_steps_for_decay, decay_rate],
+                   name, "StepTimeDecay") as name:
+    learning_rate = ops.convert_to_tensor(learning_rate, name="learning_rate")
+    dtype = learning_rate.dtype
+    decay_rate = cast(decay_rate, dtype)
+    if mod(epoch, num_steps_for_decay).eval():
```

Comment:
Ok, I just thought if I was working with pure numbers, I don't need to go through the compute mechanism.  I'll change this too.